android/java/org/chromium/chrome/browser/crypto_wallet/fragments/dapps/DAppsMessageFragment.java

```diff
@@ -37,8 +49,28 @@ public View onCreateView(
         mSignMessageText = view.findViewById(R.id.sign_message_text);
 
         mUnicodeEscapeVersion = false;
-        updateText(mUnicodeEscapeVersion);
-        if (Validations.hasUnicode(mCurrentSignMessageRequest.message)) {
+        final String message;
+        final boolean isEip712;
+        if (mCurrentSignMessageRequest.signData.which() == SignDataUnion.Tag.EthStandardSignData) {
+            message = mCurrentSignMessageRequest.signData.getEthStandardSignData().message;
+            isEip712 = false;
+        } else if (mCurrentSignMessageRequest.signData.which()
+                == SignDataUnion.Tag.EthSignTypedData) {
+            message = mCurrentSignMessageRequest.signData.getEthSignTypedData().message;
+            isEip712 = true;
+        } else if (mCurrentSignMessageRequest.signData.which()
+                == SignDataUnion.Tag.SolanaSignData) {
+            message = mCurrentSignMessageRequest.signData.getSolanaSignData().message;
+            isEip712 = false;
+        } else if (mCurrentSignMessageRequest.signData.which() == SignDataUnion.Tag.EthSiweData) {
+            message = mCurrentSignMessageRequest.signData.getEthSiweData().statement;
+            isEip712 = false;
+        } else {
+            message = "";
+            isEip712 = false;
```

Comment:
Compiler doesn't like that way:
```
DAppsMessageFragment.java:78: error: local variables referenced from a lambda expression must be final or effectively final
                updateText(mUnicodeEscapeVersion, message, isEip712);
                                                           ^
1 error
```
 because `updateText` is indeed invoked inside of the lambda
 ```Java
             warningLinkText.setOnClickListener(v -> {
....
                updateText(mUnicodeEscapeVersion, message, isEip712);
            });
 ```